src/Factory/ProductFactory.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Factory;
+
+use App\Entity\CsvProduct;
+use App\Entity\ProductInterface;
+
+class ProductFactory
+{
+    public function create(array $arguments): ProductInterface
```

Comment:
This interface is hard to use as the array structure is not documented. It makes sense to split the array into 3 separate named arguments.